vm/src/function.rs

```diff
@@ -218,6 +265,32 @@ pub trait FromArgs: Sized {
     fn from_args(vm: &VirtualMachine, args: &mut PyFuncArgs) -> Result<Self, ArgumentError>;
 }
 
+/// Handling behaviour when the argument is and isn't present, used to implement OptionalArg.
+pub trait ArgHandler: Sized {
```

Comment:
I don't really see why this trait should be necessary...

Reply:
I didn't want to use the FromArgs-OptionalArg extractor as it deals with PyFuncArgs, not a single argument, which we have in this case.

Reply:
If we avoid using `OptionalArg` as I suggested in https://github.com/RustPython/RustPython/pull/755#discussion_r269729006, then I think we can go back to just using `TryFromObject` like it did before.